src/arch/mod.rs

```diff
@@ -0,0 +1,6 @@
+#[cfg(target_arch = "riscv64")]
+mod riscv64;
+#[cfg(target_arch = "riscv64")]
+pub use riscv64::new_arch;
+
+pub trait Architecture {}
```

Comment:
It would be easier if this file was named `arch.rs` inside the `src/` directory

Reply:
Why ?